### ISSUE TITLE
Garbage Collection Question

### ISSUE THREAD

**Issue by ro-savage** [Runn-Fast/deploy-to-heroku-action#9]
Does this code run on every PR creation? Every PR close? Every change? At certain intervals?

It looks like go through all the apps, and remove any that don't have PRs anymore?

https://github.com/Runn-Fast/deploy-to-heroku-action/blob/3a095dd68a8dff60270467336de1585f6b2cec9e/src/garbage-collect-heroku-apps.ts#L9-L44

**Comment by stayradiated** [comment]
The garbage collection runs whenever the workflow is triggered -- this currently happens whenever a commit is pushed to a Pull Request or the `development` / `test` / `production` branches

**Comment by ro-savage** [comment]
I guess we can't run a github action on close that would just find the closed PR and remove it?

Probably not really a big deal to Runn this all the time, as it should be fast.

**Comment by stayradiated** [comment]
> I guess we can't run a github action on close that would just find the closed PR and remove it?

I don't think we can hook into the "pull request closed" event -- but what we can do is schedule it as a separate workflow that runs once a day.

**Comment by stayradiated** [comment]
I had a read through the docs and it seems like it might actually be possible to hook into the "closed" event for pull requests. I'll try it out.

Docs: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
